### PR TITLE
MGDSTRM-9748-Update FSO to Quarkus 2.13

### LINKED ISSUE #803: Remove snakeyaml managed version when Quarkus-managed library no longer impacted by CVE-2022-25857
Remove snakeyaml managed version when Quarkus-managed library no longer impacted by https://github.com/advisories/GHSA-3mc7-4q67-w48m

### PR DESCRIPTION
**Pull request by biswassri**
Also fixes #803 removal of snakeyaml version overide

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -239,12 +235,9 @@
                 <artifactId>jakarta.validation-api</artifactId>
                 <version>${jakarta.validation.version}</version>
             </dependency>
-            <!-- CVE-2022-25857, CVE-2022-38752 => should be removed once Quarkus uses it out of the box -->
-            <!-- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/issues/803 covers the removal of this override -->
             <dependency>
                 <groupId>org.yaml</groupId>
                 <artifactId>snakeyaml</artifactId>
-                <version>${snakeyaml.version}</version>
             </dependency>
```
Comment (MikeEdgar) [inline pom.xml:241]: I think this dependency can be removed entirely. We don't use it directly and it will be pulled in via Jackson.

**Review by MikeEdgar — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (biswassri) [reply pom.xml:241]: Sure. I'll do that.

**Review by MikeEdgar — APPROVED** [review]
lgtm

**Comment by MikeEdgar** [comment]
@shawkins , do we want to update to JOSDK 4.0.9 and fabric8 5.12.4 with this? There are code changes for the JOSDK update.

https://github.com/quarkiverse/quarkus-operator-sdk/releases/tag/4.0.7

**Comment by shawkins** [comment]
> @shawkins , do we want to update to JOSDK 4.0.9 and fabric8 5.12.4 with this? There are code changes for the JOSDK update.

Yes. Generally they all need to move in lock step.

**Comment by biswassri** [comment]
Sure! I'll update these dependencies also.

**Comment by rareddy** [comment]
Do we not have an automatic dependency checker maven plugin for FSO?

**Comment by biswassri** [comment]
> Do we not have an automatic dependency checker maven plugin for FSO?

Yes, but I don't think it's still configured completely yet. This was the PR https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/pull/848

**Review by rareddy — APPROVED** [review]
lgtm

**Comment by biswassri** [comment]
@MikeEdgar @shawkins I will open another PR with the JOSDK and fabric8 upgrades and merge this one.

**Pull request merged**